OWML.ModHelper.Menus/ModLayoutButton.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System;
+using System.Reflection;
+using OWML.Common.Menus;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace OWML.ModHelper.Menus
+{
+    public class ModLayoutButton : IModLayoutButton
+    {
+        public event Action OnClick;
+        public Button Button { get; }
+        public IModMenu Menu { get; private set; }
+        public HorizontalLayoutGroup LayoutGroup { get; private set; }
+
+        private int _index;
+        private readonly UIStyleApplier _buttonStyleApplier;
+        private readonly FieldInfo _texts;
+        private readonly FieldInfo _foregrounds;
+
+        public int Index
+        {
+            get => Button.transform.parent == null ? _index : Button.transform.GetSiblingIndex();
+            set
+            {
+                _index = value;
+                Button.transform.SetSiblingIndex(value);
+            }
+        }
+
+        public ModLayoutButton(Button button, IModMenu menu)
+        {
+            Button = button;
+            Button.onClick.AddListener(() => OnClick?.Invoke());
+            GameObject.Destroy(Button.GetComponentInChildren<Text>().gameObject);
+            var layoutObject = new GameObject("LayoutGroup", new Type[] { typeof(RectTransform) });
+            layoutObject.transform.SetParent(button.transform);
+            var target = layoutObject.AddComponent<Image>();
+            target.raycastTarget = true;
+            target.color = new Color(255, 255, 255, 0);//transparent
```

Comment:
I think you could use Color.clear or something similar

Reply:
I really should have just looked at the list, done now